### PR TITLE
Add new FC980C controller

### PR DESCRIPTION
**Pull request by wlhlm**
Add support for a new ARM-based controller for the Leopold FC980C (designed by me).

## Description

Before adding the config for the new controller, this PR first moves the existing QMK support for TMK's Leopold controllers under the `keyboards/leopold` path. This is the reason this PR targets the develop branch.

More information on the controller (including sources) can be found [here](https://github.com/wlhlm/fc980c-controller).

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/leopold/fc980c/wlhlm/actuation_point.c [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+#include "actuation_point.h"
+
+#include "i2c_master.h"
+#include "debug.h"
+#include "util.h"
+#include "wait.h"
+
+#include QMK_KEYBOARD_H
+
+#define ACTUATION_POINT_I2C_TIMEOUT 100
```
Comment (drashna) [inline keyboards/leopold/fc980c/wlhlm/actuation_point.c:14]: just so it's configurable unless there is a specific reason for not
```suggestion
#ifndef ACTUATION_POINT_I2C_TIMEOUT
#    define ACTUATION_POINT_I2C_TIMEOUT 100
#endif
```

File: keyboards/leopold/fc980c/wlhlm/keymaps/default/keymap.c [hunk 2]
```diff
@@ -0,0 +1,38 @@
+// Copyright 2023 Wilhelm Schuster
```
Comment (drashna) [inline keyboards/leopold/fc980c/wlhlm/keymaps/default/keymap.c:1]: Since the main folder has a default keymap, not sure that this one needs it,

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/leopold/fc980c/wlhlm/keymaps/default/keymap.c (hunk 2 above)
Comment (wlhlm) [reply keyboards/leopold/fc980c/wlhlm/keymaps/default/keymap.c:1]: I'm sorry, I don't quite understand. Do you mean the default keymap in fc980c/tmk?

**Comment by wlhlm** [comment]
Rebased. I'm still not sure what "main folder default keymap" is referring to.

File: keyboards/leopold/fc660c/fc660c.c [hunk 3]
```diff
@@ -14,7 +14,7 @@ GNU General Public License for more details.
 You should have received a copy of the GNU General Public License
 along with this program.  If not, see <http://www.gnu.org/licenses/>.
 */
-#include "fc660c.h"
+#include QMK_KEYBOARD_H
```
Comment (fauxpark) [inline keyboards/leopold/fc660c/fc660c.c:17]:
```suggestion
#include "fc660c.h"
```

**Comment by keyboard-magpie** [comment]
> Rebased. I'm still not sure what "main folder default keymap" is referring to.

Drashna is referring to keyboards/leopold/fc980c/tmk/keymaps/default/keymap.c

File: keyboards/leopold/fc660c/readme.md [hunk 4]
```diff
@@ -11,13 +11,13 @@ This file and the entire QMK port of this board are based on its TMK implementat
...
 
-    make fc660c:default
+    make leopold/fc660c:default
 
 See [build environment setup](https://docs.qmk.fm/#/getting_started_build_tools) then the [make instructions](https://docs.qmk.fm/#/getting_started_make_guide) for more information.
 
 To flash the firmware, press the reset button on the controller and run:
 
-    sudo make fc660c:default:dfu
+    sudo make leopold/fc660c:default:dfu
```
Comment (fauxpark) [inline keyboards/leopold/fc660c/readme.md:20]:
```suggestion
    make leopold/fc660c:default:flash
```

File: keyboards/leopold/fc980c/tmk/readme.md [hunk 5]
```diff
@@ -13,13 +13,13 @@ This file and the entire QMK port of this board are based on its TMK implementat
...
 
-    make fc980c:default
+    make leopold/fc980c/tmk:default
 
 See [build environment setup](https://docs.qmk.fm/#/getting_started_build_tools) then the [make instructions](https://docs.qmk.fm/#/getting_started_make_guide) for more information.
 
 To flash the firmware, press the reset button on the controller and run:
 
-    sudo make fc980c:default:dfu
+    sudo make leopold/fc980c/tmk:default:dfu
```
Comment (fauxpark) [inline keyboards/leopold/fc980c/tmk/readme.md:22]:
```suggestion
    make leopold/fc980c/tmk:default:flash
```

File: keyboards/leopold/fc980c/tmk/tmk.c [hunk 6]
```diff
@@ -15,7 +15,7 @@ You should have received a copy of the GNU General Public License
 along with this program.  If not, see <http://www.gnu.org/licenses/>.
 */
 
-#include "fc980c.h"
+#include QMK_KEYBOARD_H
```
Comment (fauxpark) [inline keyboards/leopold/fc980c/tmk/tmk.c:18]:
```suggestion
#include "tmk.h"
```

File: keyboards/leopold/fc980c/wlhlm/info.json [hunk 7]
```diff
@@ -0,0 +1,158 @@
...
+    "build": {
+        "lto": true
+    },
+
+    "matrix_size": {
+        "rows": 8,
+        "cols": 16
+    },
+
+    "debouce": 0,
```
Comment (fauxpark) [inline keyboards/leopold/fc980c/wlhlm/info.json:40]:
```suggestion
    "debounce": 0,
```

File: keyboards/leopold/fc980c/wlhlm/info.json [hunk 8]
```diff
@@ -0,0 +1,158 @@
...
+
+    "build": {
+        "lto": true
+    },
+
+    "matrix_size": {
+        "rows": 8,
+        "cols": 16
+    },
+
```
Comment (fauxpark) [inline keyboards/leopold/fc980c/wlhlm/info.json:39]:
```suggestion
```

File: keyboards/leopold/fc980c/wlhlm/keymaps/actuation_point_example/keymap.c [hunk 9]
```diff
@@ -0,0 +1,80 @@
+// Copyright 2023 Wilhelm Schuster
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#include <stdio.h>
+
+#include "debug.h"
+
```
Comment (fauxpark) [inline keyboards/leopold/fc980c/wlhlm/keymaps/actuation_point_example/keymap.c:7]:
```suggestion
```

File: keyboards/leopold/fc980c/wlhlm/keymaps/default/keymap.c [hunk 10]
```diff
@@ -0,0 +1,38 @@
...
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#include "debug.h"
+
```
Comment (fauxpark) [inline keyboards/leopold/fc980c/wlhlm/keymaps/default/keymap.c:5]:
```suggestion
```

File: keyboards/leopold/fc980c/wlhlm/wlhlm.c [hunk 11]
```diff
@@ -0,0 +1,39 @@
+// Copyright 2023 Wilhelm Schuster
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#include QMK_KEYBOARD_H
+
+#include "debug.h"
+#include "i2c_master.h"
+
+#include "actuation_point.h"
+
```
Comment (fauxpark) [inline keyboards/leopold/fc980c/wlhlm/wlhlm.c:10]:
```suggestion
```

File: keyboards/leopold/fc980c/wlhlm/wlhlm.h [hunk 12]
```diff
@@ -0,0 +1,15 @@
+// Copyright 2023 Wilhelm Schuster
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#pragma once
+
+#include "quantum.h"
```
Comment (fauxpark) [inline keyboards/leopold/fc980c/wlhlm/wlhlm.h:6]:
```suggestion
#include "quantum.h"
#include "actuation_point.h"
```

File: keyboards/leopold/fc980c/wlhlm/keymaps/default/keymap.c [hunk 13]
```diff
@@ -0,0 +1,38 @@
...
+#include QMK_KEYBOARD_H
+
+#include "actuation_point.h"
+
```
Comment (fauxpark) [inline keyboards/leopold/fc980c/wlhlm/keymaps/default/keymap.c:9]:
```suggestion
```

File: keyboards/leopold/fc980c/wlhlm/keymaps/actuation_point_example/keymap.c [hunk 14]
```diff
@@ -0,0 +1,80 @@
...
+#include QMK_KEYBOARD_H
+#include "actuation_point.h"
```
Comment (fauxpark) [inline keyboards/leopold/fc980c/wlhlm/keymaps/actuation_point_example/keymap.c:9]:
```suggestion
#include QMK_KEYBOARD_H
```

File: keyboards/leopold/fc980c/wlhlm/info.json [hunk 15]
```diff
@@ -0,0 +1,158 @@
...
+    "debouce": 0,
+
+    "indicators": {
+        "num_lock": "A3",
+        "caps_lock": "A2",
+        "scroll_lock": "A1",
+        "on_state": 1
```
Comment (fauxpark) [inline keyboards/leopold/fc980c/wlhlm/info.json:46]:
```suggestion
```

File: keyboards/leopold/fc980c/wlhlm/info.json [hunk 16]
```diff
@@ -0,0 +1,158 @@
...
+        "vid": "0xFEED",
+        "pid": "0xF980",
+        "device_version": "2.0.0",
+
+        "max_power": 100
+    },
+
+    "processor": "STM32F401",
+    "bootloader": "stm32-dfu",
+    "board": "GENERIC_STM32_F401XC",
```
Comment (fauxpark) [inline keyboards/leopold/fc980c/wlhlm/info.json:17]:
```suggestion
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Pull request closed without merging**